yt_dlp/extractor/bilibili.py

```diff
@@ -589,14 +690,40 @@ def _real_extract(self, url):
                     '__post_extractor': self.extract_comments(aid),
                 })
         else:
-            return {
-                **metainfo,
-                'duration': float_or_none(play_info.get('timelength'), scale=1000),
-                'chapters': self._get_chapters(aid, cid),
-                'subtitles': self.extract_subtitles(video_id, cid),
-                'formats': self.extract_formats(play_info),
-                '__post_extractor': self.extract_comments(aid),
-            }
+            formats = self.extract_formats(play_info)
+
+            if not traverse_obj(play_info, ('dash')):  # for legacy-only formats
+                has_qn = lambda x: str_or_none(x) in traverse_obj(formats, (..., 'format_id'))
+                for qn in traverse_obj(play_info, ('accept_quality', lambda _, v: not has_qn(v), {int})):
+                    formats.extend(traverse_obj(
+                        self.extract_formats(self._get_play_url(video_id, cid, headers=headers, qn=qn)),
+                        (lambda _, v: not has_qn(v.get('format_id')))))
+                if traverse_obj(formats, lambda _, v: v['fragments']):
+                    # choose best quality for multi_video
+                    formats = [max(formats, key=lambda v: v.get('quality', 0))]
```

Comment:
It seems that even for those with flv format, a single-file mp4 format is still available.
```suggestion
                if traverse_obj(formats, lambda _, v: v['fragments']):
                    # choose best quality for multi_video
                    formats = traverse_obj(formats, lambda _, v: not v.get('fragments')) or formats
```
Given these old videos don't have high-quality options anyway, dropping multi-video flv might be a better choice (while keeping fallback in case of flv-only scenario)?